iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/convert.inl

```diff
@@ -91,7 +91,7 @@ inline bool convert::fromString<char>(const char* v, char& dest) noexcept
 template <>
 inline bool convert::fromString<string<100>>(const char* v, string<100>& dest) noexcept
 {
-    dest = string<100>(TruncateToCapacity, v);
+    dest = string<100>(TruncateToCapacity, v, strnlen(v, string<100>::capacity()));
```

Comment:
@elfenpiff @FerdinandSpitzschnueffler `convert.hpp` will very likely be moved to `iceoryx_dust`.